From c3c578f2b673b8f6eff5079924d54a06d4a2512a Mon Sep 17 00:00:00 2001
From: "Poluri, Sarat Chandra" <sarat.chandra.poluri@intel.com>
Date: Tue, 9 Apr 2019 22:50:51 +0000
Subject: [PATCH 1381/1443] media: ici: Fix memory leak in driver due to
 virt_dev_id not being initialized properly.

Without virt_dev_id being initialized to -1, the default value will be zero.
Zero is a valid virt_dev_id value and will cause unmap_buf_virt to be called
instead of unmap_buf.

Change-Id: I2dc6c7ef169ed3a3fe1c77119cb98b428064a2a8
Signed-off-by: Poluri, Sarat Chandra <sarat.chandra.poluri@intel.com>
---
 drivers/media/pci/intel/ici/ici-isys-stream-device.c | 1 +
 drivers/media/pci/intel/ici/ici-isys-stream.c        | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ici/ici-isys-stream-device.c b/drivers/media/pci/intel/ici/ici-isys-stream-device.c
index 9633698..98c839b 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream-device.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream-device.c
@@ -319,6 +319,7 @@ int stream_device_register(struct ici_stream_device *strm_dev)
 		return -EINVAL;
 	}
 	strm_dev->minor = -1;
+	strm_dev->virt_dev_id = -1;
 
 	cdev_init(&strm_dev->cdev, &ici_stream_fops);
 	strm_dev->cdev.owner = ici_stream_fops.owner;
diff --git a/drivers/media/pci/intel/ici/ici-isys-stream.c b/drivers/media/pci/intel/ici/ici-isys-stream.c
index 09fcd02..1384fc3 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream.c
@@ -1165,8 +1165,6 @@ static int stream_fop_open(struct inode *inode, struct file *file)
 
 	mutex_unlock(&isys->mutex);
 
-	strm_dev->virt_dev_id = -1;
-
 	return 0;
 
 out_lib_init:
-- 
2.7.4

